docarray/typing/tensor/audio/abstract_audio_tensor.py

```diff
@@ -0,0 +1,25 @@
+from abc import ABC, abstractmethod
+from typing import BinaryIO, TypeVar, Union
+
+from docarray.typing.tensor.abstract_tensor import AbstractTensor
+
+T = TypeVar('T', bound='AbstractAudioTensor')
+
+
+class AbstractAudioTensor(AbstractTensor, ABC):
+    @abstractmethod
+    def save_to_wav_file(
+        self: 'T',
+        file_path: Union[str, BinaryIO],
+        sample_rate: int = 44100,
+        sample_width: int = 2,
+    ) -> None:
+        """
+        Save tensor to a .wav file. Mono/stereo is preserved.
+
+        :param file_path: path to a .wav file. If file is a string, open the file by
+            that name, otherwise treat it as a file-like object.
+        :param sample_rate: sampling frequency
+        :param sample_width: sample width in bytes
+        """
+        ...
```

Comment:
I suggest the following:
* put all audio-specific logic (save_to_wav_file without tensor to byte conversion and _to_node_protobuf) into 1 class, ideally this class. I think we can make these 2 methods concrete and therefore this class can be turned into a mixin rather than an abstract class.
* move the tensor-to-bytes logic into tensor classes as a method `def tobytes()`. Then use this common method when implementing the common `save_to_wav_file()` method. Alternatively, and if you believe this logic is more related to audio, you can define `to_audio_bytes` in concrete audio tensor classes, but still the implementation of `save_to_wav_file` should stay common in the parent audio tensor class.
* A concrete audio tensor class is just a class that combines the base audio tensor class and the concrete.
I will be making suggestions in this direction